yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1055,67 +1056,156 @@ def pixelize_sph_kernel_projection(
                     # now we just use the kernel projection
                     buff[xi, yi] +=  prefactor_j * itab.interpolate(q_ij2)
 
-@cython.initializedcheck(True)
-@cython.boundscheck(True)
+@cython.boundscheck(False)
 @cython.wraparound(False)
-@cython.cdivision(True)
-def interpolate_sph_arbitrary_positions_gather(
-        np.float64_t[:] buff,
-        np.float64_t[:, :] particle_positions,
-        np.float64_t[:, :] interpolation_positions,
-        np.float64_t[:] hsml, np.float64_t[:] pmass,
-        np.float64_t[:] pdens,
-        np.float64_t[:] quantity_to_smooth,
-        PyKDTree kdtree=None, int use_normalization=1,
-        kernel_name="cubic", pbar=None, int num_neigh=32):
+def interpolate_sph_gather(np.float64_t[:] buff,
```

Comment:
maybe call this `interpolate_sph_positions_gather` or just leave the old name? Alternatively, maybe add a one-two sentence docstring to this function and `interpolate_sph_grid_gather` explaining the difference between what they do?